### PR TITLE
add Embedded Challenge

### PR DESCRIPTION
**Pull request by phoddie**
First attempt to integrate the Embedded Challenge article here.

Since it goes with the original challenge, I created a directory to hold both so that they appear together in the side-bar. If you have another idea for organizing these, that's fine too.

The markdown renders more-or-less reasonably. The only detail is that I'd prefer the thumbnails to be scaled to 75%. My attempts to convince Astro and markdown to cooperate to make that happen. Maybe some else knows how to do that?

### CODE REVIEW

File: src/content/docs/Challenge/challenge-embedded.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+---
+
+<link rel="stylesheet" href="../index.css">
+
+The Embedded Challenge is a port of this site's [Web Challenge](https://hardenedjs.org/challenge/challenge) to [Moddable Six](https://www.moddable.com/moddable-six), an embedded device powered by the popular [ESP32-S3](https://www.espressif.com/en/products/socs/esp32-s3) microcontroller. The Embedded Challenge runs on the XS JavaScript engine. XS implements Hardened JavaScript natively, as part of the engine itself, including `lockdown()` and the `Compartment` class. The XS implementation of Hardened JavaScript is a completely separate implementation from the SES shim used by web-based Challenge.
+
+## Running the Embedded Challenge
+
+To run the Embedded Challenge, you'll need a Moddable Six (or equivalent) and a computer (macOS, Windows, or Linux) with the Moddable SDK and ESP-IDF installed. From there, it is straightforward to build, install, and run:
+
```
Comment (kriskowal) [inline src/content/docs/Challenge/challenge-embedded.md:23]: Could use a tip on where to find the Moddable repo to clone and set up the MODDABLE environment variable.

File: src/content/docs/Challenge/challenge-embedded.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+The Embedded Challenge also runs on macOS, Windows, and Linux using the `mcsim`, the Moddable SDK's simulator. This allows you to try out the Embedded Challenge while waiting for your embedded device to be delivered. To run the Embedded Challenge in the simulator, install the Moddable SDK and then:
+
+```
+> cd $MODDABLE/contributed/challenge
+> mcconfig -d -m -p sim/moddable_six
+```
+
+![mcsim](../../../assets/embedded-challenge-simulator.png)
+
+The application works exactly the same. However, BLE is unavailable so you cannot download your own attack scripts. You can edit the source code of built-in scripts, however, to try various attacks. While running in the simulator, the xsbug JavaScript debugger is available to debug your attacks and explore the runtime environment.
```
Comment (kriskowal) [inline src/content/docs/Challenge/challenge-embedded.md]: XSBug could use a link or instructions on how to build and run it.

**Review by kriskowal — APPROVED** [review]
Sizes looked good to me locally. A couple elaborations might go a long way to get folks thru to the end.

File: src/content/docs/Challenge/challenge-embedded.md (hunk 2 above)
Comment (phoddie) [reply src/content/docs/Challenge/challenge-embedded.md]: Will add a link. It is built as part of setting up the Moddable SDK, so nothing special there.

File: src/content/docs/Challenge/challenge-embedded.md (hunk 1 above)
Comment (phoddie) [reply src/content/docs/Challenge/challenge-embedded.md:23]: Good point. Will do.

**Review by kriskowal — APPROVED** [review]

**Pull request merged**